charts/redpanda/templates/statefulset.yaml

```diff
@@ -321,6 +321,30 @@ spec:
             limits:
               cpu: {{ .Values.resources.cpu.cores }}
               memory: {{ .Values.resources.memory.container.max }}
+      {{- if .Values.statefulset.sideCars.configWatcher.enabled }}
+        - name: config-watcher
+          image: {{ .Values.image.repository }}:{{ template "redpanda.tag" . }}
+          command:
+            - /bin/bash # could be expanded for multiple scripts
+            - -c 
+            - /etc/secrets/config-watcher/scripts/sasl-user.sh
+          volumeMounts:
+            - name: {{ template "redpanda.fullname" . }}-config-watcher
+              mountPath: /etc/secrets/config-watcher/scripts
+              readOnly: true
+          {{- if and .Values.auth.sasl.enabled (not (empty .Values.auth.sasl.secretRef )) }}
+            - name: {{ .Values.auth.sasl.secretRef }}
+              mountPath: /etc/secrets/users
+              readOnly: true
+          {{- end }}
```

Comment:
Will this be even allowed to start from sasl disabled to then move to sasl enabled? If I recall stateful set will not allow to change volume and volumeMounts definition.

Reply:
this isn't a change to volumes being mounted, it is an addition to the new container so it can mount the same volume. And yes the new changes alllow us to move from disabled to enabled which was a known issue before.